components/ledger/pkg/ledger/query/worker.go

```diff
@@ -33,14 +53,18 @@ func (w *Worker) Run(ctx context.Context) error {
 			logging.FromContext(ctx).Debugf("CQRS worker stopped")
 			close(stopChan)
 			return nil
-		case <-time.After(w.Interval):
-			if err := w.run(ctx); err != nil {
+		case log := <-w.logChan:
+			if err := w.processLog(ctx, log.store, log.log); err != nil {
 				if err == context.Canceled {
 					logging.FromContext(ctx).Debugf("CQRS worker canceled")
 					return err
 				}
 				logging.FromContext(ctx).Errorf("CQRS worker error: %s", err)
```

Comment:
In case of error, i think we should not update the next log id. Either we crash the app to allow the polling part to make its job, either the process need a way to retry later.

Reply:
Instead of crashing the app, what do you think about crashing the worker goroutine and relaunching it ?